src/common/cockpitlocale.c

```diff
@@ -103,6 +102,7 @@ cockpit_locale_set_language (const gchar *value)
       g_setenv ("LANG", locale, TRUE);
     }
 
-  strncpy (previous, locale, sizeof (previous));
+  strncpy (previous, value, sizeof (previous) - 1);
```

Comment:
See the documentation. "except that at most n bytes of src are copied". If value is shorter, it'll copy the shorter value and a null terminator.